### PR TITLE
PDFjs now compatible with Librejs

### LINKED ISSUE #3775: PDF.js is not compatible with LibreJS
Being that pdf.js is free software, it would be in the best interests of the pdf.js team to tag the licenses of all Javascript files part of the project in a way that computers can read and confirm that users are in fact allowed to change the code. A practical implication of this is that pdf.js will no longer be blocked by the LibreJS browser addon.

For an example of how this has been done in a similar web application, see https://github.com/ether/etherpad-lite/pull/1066. For instructions by the LibreJS team, see http://www.gnu.org/software/librejs/free-your-javascript.html.

### PR DESCRIPTION
**Pull request by himanish-star**
I hope this fixes #3775 .

I have made the changes in the files mentioned by @yurydelendik 
1. src/pdf.js
2. web/pdf_page_view.js
3. web/viewer.js
**web/compatibility.js (I couldn't find this file. I may have gone wrong on my part. So if someone could locate this file for me I would fix this too)**

### CODE REVIEW

File: src/pdf.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
-/* Copyright 2012 Mozilla Foundation
+/*
+ * @licstart  The following is the entire license notice for the
+ *  JavaScript code in this page.
```
Comment (Snuffleupagus) [inline src/pdf.js]: Why is the word `JavaScript` indented with one space here?

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
First of all, the list in https://github.com/mozilla/pdf.js/issues/3775#issuecomment-157149664 is more than two years old, hence it's no longer current.

*Edit:* I'm also guessing that you didn't actually test these changes by building PDF.js, e.g. using `gulp generic` locally, and check the resulting files (since the licence headers you modified are all stripped out during the build step)!?

In general, I do *not* think that we should be editing the licence header of individual files like this, since it for one doesn't scale well.
A much better solution would be to add this during the build step instead!

I'm not sure if we should change the https://github.com/mozilla/pdf.js/blob/master/src/license_header.js file. It might be better to add a new one e.g. `license_header_libre.js` and use that in https://github.com/mozilla/pdf.js/blob/master/gulpfile.js instead.

**Comment by himanish-star** [comment]
@Snuffleupagus , I have tried to make the changes requested. I have replaced the existing usages of license_header.js with license_header_libre.js. I made these replacements instead of using both as using both will lead to two license headers. Thus I have replaced the former one with the latter one.

File: gulpfile.js [hunk 2]
```diff
@@ -127,7 +127,9 @@ function createWebpackConfig(defines, output) {
     BUNDLE_VERSION: versionInfo.version,
     BUNDLE_BUILD: versionInfo.commit,
   });
-  var licenseHeader = fs.readFileSync('./src/license_header.js').toString();
+  var licenseHeaderLibre = fs
+                           .readFileSync('./src/license_header_libre.js')
+                           .toString();
```
Comment (Snuffleupagus) [inline gulpfile.js]: This isn't how we usually break lines in PDF.js, let's use to following format instead:
```javascript
  var licenseHeaderLibre =
    fs.readFileSync('./src/license_header_libre.js').toString();
```

File: gulpfile.js [hunk 3]
```diff
@@ -1012,7 +1014,9 @@ gulp.task('lib', ['buildnumber'], function () {
       'pdfjs-lib': '../pdf',
     },
   };
-  var licenseHeader = fs.readFileSync('./src/license_header.js').toString();
+  var licenseHeaderLibre = fs
+                           .readFileSync('./src/license_header_libre.js')
+                           .toString();
```
Comment (Snuffleupagus) [inline gulpfile.js]: The formatting comment above applies here as well.

**Review by Snuffleupagus — APPROVED** [review]
That looks much better, thanks!

@yurydelendik, @timvandermeij Would you agree that this is a reasonable way to address the issue?

File: gulpfile.js (hunk 2 above)
Comment (himanish-star) [reply gulpfile.js]: Yup , I get it . Working on the indentation now

**Review by timvandermeij — APPROVED** [review]
Looks like a good solution to me. If we are not using the regular `license_header.js` file anymore, I'm also fine with changing that one instead of creating a new one, but either way is fine by me. I'll let @yurydelendik decide on that one.

**Comment by Snuffleupagus** [comment]
> I'm also fine with changing that one instead of creating a new one, but either way is fine by me.

Since every file should have a licence header, and we're linking to that file from https://github.com/mozilla/pdf.js/wiki/Style-Guide, it thus seemed like a good idea to keep it around.

**Comment by timvandermeij** [comment]
Nice, I had not noticed that one yet. It's indeed good to have that in the repository. Let's keep it this way then. If Yury is also fine with the current diff, we can land it.

**Comment by himanish-star** [comment]
@Snuffleupagus , @timvandermeij any updates ?

**Comment by timvandermeij** [comment]
I have looked into this a bit more and this is indeed all logic we use for the license headers in built files, so let's merge this.

**Comment by timvandermeij** [comment]
Thank you for the patch!

**Pull request merged**